src/pages/index.js

```diff
@@ -65,55 +45,94 @@ function handleEditProfileButtonClick() {
   formValidators["profile-form"].disableSubmitButton();
 }
 
-profilePopup.setEventListeners();
+function loadApiCards() {//no longer used save as a part of the "emergency option" so holding on to just in case
+  api.getCards().then(data => {
+    data.forEach((datum) => (renderCard(datum)));
+  });
+}
 
-editProfileButton.addEventListener("click", handleEditProfileButtonClick);
+function restoreOriginalCards() {//in case I ever go nuclear on the cards again by accident... call on standby at the bottom
+  initialCards.forEach((card) => {
+    api.postCard(card);    
+  });
+  loadApiCards();
+}
+
+function handleDelButtonClick(card) {
+  deleteDialogue.open();
+  deleteIdentification = card;
+}
 
-function renderCard(card) {//to address the recursive code I had been writing in the section and addphoto initializations
-  const newCard = new Card(card, "#card", handlePhotoClick);
+function renderCard(card) {
+  const newCard = new Card(card, "#card", handlePhotoClick, handleDelButtonClick, handleLikeClick);//ammended the delete and like handler so it could tell the other popup to open when the Card's del button was pressed and utilize the api.  
   const newCardElement = newCard.generateCardElement();
   cardSection.addItem(newCardElement);
 }
 
-const addPopup = new PopupWithForm(addPhotoPopup, {
-  handleFormSubmit: ({ title, image_url: imageUrl }) => {
-    const newObj = {
-      name: title,
-      link: imageUrl
-    };
-    renderCard(newObj);    
-    formValidators["add-photo-form"].resetFormValidation();
-    addPopup.close();
-  },
-});
-
-
-addPopup.setEventListeners();
-
-const cardSection = new Section(
-{
-  items: initialCards,
-  renderer: renderCard,
-},
-".cards"
-);
-
-cardSection.renderItems();
+function handleDeleteSubmit() {//did this as a pass-to for the delete popup just because I was having such a difficult time workshopping how exactly to delete, needed to segregate it. Came up with this.
+  api.deleteCard(deleteIdentification.id)
+  .then(() => {
+    deleteDialogue.renderLoading(true, "Deleting...");    
+    deleteIdentification.deleteElement();//I guess I was storing the to-be-deleted card all along and just needed to access it. WOW.
+    formValidators["delete-photo-form"].disableSubmitButton();
```

Comment:
Hey there

A question- I know the delete form doesn't have inputs but I thought it would be a good idea to disable the submit it to prevent the user from double submitting and causing havoc that way. Is this a valid concern or is it a non-issue? (Or, in other words, I don't really know what would happen if the submission event was made twice in a row while the api call was still working for this form- can you explain it to me what would happen?)